app/policies/department_policy.rb

```diff
@@ -20,6 +20,29 @@ def initialize(user, department)
     @department = department
   end
 
+  def permitted_attributes
+    return [user_ids: [], contact_attributes: contact_attrs] if @user.superadmin?
+    return [contact_attributes: contact_attrs] if @user.department_manager?
+    []
+  end
+
+  def phone_attrs
+    [:id, :body, :label, :_destroy, :type, :contacts_id]
+  end
+
+  def email_attrs
+    [:id, :body, :label, :_destroy, :type, :contacts_id]
+  end
+
+  def contact_attrs
+    [
+      :id, :name, :_destroy, :contactable_id, :contactable_type, :street,
+      :extended, :city, :postal_code,
+      contact_emails_attributes: email_attrs,
+      contact_phones_attributes: phone_attrs
+    ]
+  end
+
```

Comment:
In the policy I can assign different permited parameters based on the logged in user role. So in this case the department manager is not allowed to set user_ids to the department, but the superadmin is.